### DISCUSSION TITLE
Orange Pi 5

### DISCUSSION THREAD

**Discussion by mschirrmeister** [kwankiu/archlinux-installer-rock5#63] [General]
In the supported devices list I saw **testers** are needed for **Orange Pi 5** models. Is there a list what exactly should be tested?
I used the image from the release section for the `Orange Pi 5`. Installation and access works fine with the standard 6.1 bsp kernel.

It did no longer boot after upgrading to the latest `6.11.0-rc6`. Unfortunately nothing on the screen.
I will run some more tests next week and try the **7Ji** kernel.

**Comment by kwankiu** [comment]
> I saw **testers** are needed for **Orange Pi 5** models

For recently released models like the Orange Pi 5 Pro / Max and the CM5.

> upgrading to the latest `6.11.0-rc6`

What is the kernel package name exactly you installed? HDMI output for RK3588 on mainline/non-BSP kernel is only available on `linux-aarch64-rockchip-armbian-git` and `linux-aarch64-rockchip-joshua-git` (only on some of the boards)

**Reply by mschirrmeister** [reply]
I don't have the **Pro** or **Max**. Only a **Plus** which runs Armbian as my NAS. Cannot mess around to much.

I think I installed `linux-aarch64-rockchip-joshua-git`. But since it did not longer came back up, I could not check further. I will come back next week with more information. 

I have not seen `linux-aarch64-rockchip-armbian-git` available via pacman. Do I need to add another repo?

**Reply by mschirrmeister** [reply]
Here is more from my tests. Everything works fine on the **Opi 5** with the default/recommended kernel `linux-aarch64-rockchip-bsp6.1-joshua-git`

```
root@sperber ~# pacman -Qe | grep linux
archlinuxarm-keyring 20240419-1
linux-aarch64-rockchip-bsp6.1-joshua-git 6.1.75.r1272327.g7348ed5d-1
linux-aarch64-rockchip-bsp6.1-joshua-git-headers 6.1.75.r1272327.g7348ed5d-1
linux-firmware-joshua-git 20240804.794da26.20240909.552ed9b8-1

root@sperber ~# uname -a
Linux sperber.home.marco.cx 6.1.75-rockchip-g7348ed5de414-dirty #1 SMP Mon Sep  9 05:44:00 UTC 2024 aarch64 GNU/Linux
```

From a base install I tried to upgrade to 3 different kernels. Below are the details.

1. Tried to install kernel `linux-aarch64-7ji`
- Installed via `acu install linux-aarch64-7ji`

After reboot the system does not come up.
Board green LED blinks, network only green led blinks. Since HDMI does not work, I cannot check locally.

2. Tried to install kernel `linux-aarch64-rockchip-joshua-git`
- Installed via `acu install linux-aarch64-rockchip-joshua-git`

After reboot the system does not come up
Board green LED blinks, network only green led blinks. Since HDMI does not work, I cannot check locally.

3. Tried to install kernel `linux-aarch64-rockchip-armbian-git`
- Build kernel linux-aarch64-rockchip-armbian-git from repo https://github.com/7Ji-PKGBUILDs/linux-aarch64-rockchip-armbian-gi
- Created local repo
- Added package `linux-aarch64-rockchip-armbian-git-6.10.8.r36.d801774.4d5b5e-1-aarch64.pkg.tar.xz` to local repo
- Installed package via `acu install linux-aarch64-rockchip-armbian-git`

After reboot the system does not come up.
Board green LED blinks, network only green led blinks. Since HDMI does not work, I cannot check locally.

**Reply by mschirrmeister** [reply]
I did another test with the newest `6.11.0` kernel and also connected a serial console. It looks like the sdcard device is not available/visible. It cannot mount the partition.

```
root@sperber ~# acu install linux-aarch64-rockchip-joshua-git

root@sperber ~# pacman -Qe | grep linux
archlinuxarm-keyring 20240419-1
linux-aarch64-rockchip-bsp6.1-joshua-git 6.1.75.r1272328.g1285ccbb-1
linux-aarch64-rockchip-bsp6.1-joshua-git-headers 6.1.75.r1272328.g1285ccbb-1
linux-aarch64-rockchip-joshua-git 6.11.r1296723.g375956b4-1
linux-firmware-joshua-git 20240804.794da26.20240909.552ed9b8-1
```

Console after the reboot

```
Device 0: unknown device
switch to partitions #0, OK
mmc1 is current device
Scanning mmc 1:1...
Found /extlinux/extlinux.conf
Retrieving file: /extlinux/extlinux.conf
reading /extlinux/extlinux.conf
382 bytes read in 4 ms (92.8 KiB/s)
1:	linux-aarch64-rockchip-joshua-git
Retrieving file: /initramfs-linux-aarch64-rockchip-joshua-git.img
reading /initramfs-linux-aarch64-rockchip-joshua-git.img
9351290 bytes read in 749 ms (11.9 MiB/s)
Retrieving file: /vmlinuz-linux-aarch64-rockchip-joshua-git
reading /vmlinuz-linux-aarch64-rockchip-joshua-git
43188736 bytes read in 3449 ms (11.9 MiB/s)
append: root=UUID=e102adfd-ae96-4366-9584-9f9f89610098 rw quiet splash androidboot.fwver=bl31-v1.45,uboot-07/30/2024
Retrieving file: /dtbs/linux-aarch64-rockchip-joshua-git/rockchip/rk3588s-orangepi-5.dtb
reading /dtbs/linux-aarch64-rockchip-joshua-git/rockchip/rk3588s-orangepi-5.dtb
148455 bytes read in 24 ms (5.9 MiB/s)
Fdt Ramdisk skip relocation
No misc partition
## Flattened Device Tree blob at 0x08300000
   Booting using the fdt blob at 0x08300000
   Using Device Tree in place at 0000000008300000, end 00000000083273e6
can't found rockchip,drm-logo, use rockchip,fb-logo
WARNING: could not set reg FDT_ERR_BADOFFSET.
failed to reserve fb-loader-logo memory
WARNING: could not set reg FDT_ERR_BADOFFSET.
Adding bank: 0x00200000 - 0xf0000000 (size: 0xefe00000)
Adding bank: 0x100000000 - 0x200000000 (size: 0x100000000)
Total: 12372.662/12407.794 ms

Starting kernel ...

[    1.746574] rk_gmac-dwmac fe1c0000.ethernet: Can not read property: rx_delay.
[    1.747235] rk_gmac-dwmac fe1c0000.ethernet: set rx_delay to 0x10
[    1.987750] optee firmware:optee: probe with driver optee failed with error -22
kbd_mode: KDSKBMODE: Inappropriate ioctl for device
ERROR: device 'UUID=e102adfd-ae96-4366-9584-9f9f89610098' not found. Skipping fsck.
mount: /new_root: can't find UUID=e102adfd-ae96-4366-9584-9f9f89610098.
ERROR: Failed to mount 'UUID=e102adfd-ae96-4366-9584-9f9f89610098' on real root
You are now being dropped into an emergency shell.
sh: can't access tty; job control turned off
[rootfs ~]#
```

The commands are very limited at this stage. `blkid` shows only the `mtdblock0` device and a nvme disk I have in the slot.
